rfcs/table.md

```diff
@@ -0,0 +1,321 @@
+        const {currentPage, pageSize} = this.state;
+        this.fetch(currentPage, pageSize);
+    }
+    onPageChange = (nextPage, prevPage) => {
+        const {pageSize} = this.state;
+        this.fetch(nextPage, pageSize);
+    }
+    render = () => {
+         return (
+             <Table
+                columns={columns}
+                dataSource={this.state.data}
+             />
+             <Pagination currentPage={this.state.currentPage} numPages={this.state.numPages} onPageChange={this.onPageChange}/>
+         );
+    }
+}
+```
+
+
+## Exports
+
+* `Table`
+* `StatefulTable`
+* `Tbody`
+* `Thead`
+* `TheadTr`
+* `Td`
+* `Th`
+* `Tr`
+
+## `Table` API
+
+* `data:Array<T>` - Required
+  Description of prop
+* `columns: Column` - Required
+  Description of prop
+* `render: (dataSource, columns) => $ReactNode` - Optional
+    Custom render table body.
+* `overrides - Object:` - Optional
+
+* `overrides(continued):`
+  * `Tbody` - Body of the table. Corresponds.
+  * `Thead` - Container for the table header related elements.
+  * `TheadTr` - A header row. `Tr` for the `Thead` element.
+  * `Th` - A header cell. Corresponds with the `th` html element.
+  * `Td` - A data cell in the table. Corresponds with the `td` html element.
+  * `Tr` - A data row that contains `Td` components.
+
+
+## `Column` API
+One of the Table columns prop for describing the table's columns, Column has the same API.
+* `title: string` - Required. Title of the column to display in the header.
+* `dataIndex: string` - Optional. Attribute at which to index the row data.
+* `render: <T>(<T>, index, Array<T>) => $ReactNode` - Optional.	Renderer of the table cell. The return value should be a ReactNode.
+* `sorter:Function`- Optional  Sort function for [local sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)
+* `defaultSortOrder:'ascend'|'descend'`: Only to be declared on one of the columns, this will be define the default sort on the initial set of data.
+
+## Presentation Components
+
+### `Tbody` API
+Please refer to the [tbody DOM API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody)
+
+### `Thead` API
+Please refer to the [thead DOM API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/thead)
+
+### `TheadTr` API
+Please refer to the [tr DOM API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr)
+
+### `Td` API
+Please refer to the [td DOM API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/td)
+
+### `Th` API
+Please refer to the [th DOM API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/th)
+
+### `Tr` API
+Please refer to the [th DOM API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr)
+
+
+### Dependencies
+
+Does this component depend on any 3rd party packages or other internal components?
+
+## Accessibility
+
+How can this component be used via keyboard controls?
+No keyboard shortcuts will be enabled.
```

Comment:
Might be nice to enable highlighting via keyboard? e.g. click in a cell, table gains keyboard focus, move the highlight with arrow keys. I'm ambivalent...also, this would add local state, which we might want to avoid if possible.

Reply:
https://bvaughn.github.io/react-virtualized/#/components/ArrowKeyStepper
Provided out of the box here

Reply:
Thanks for the suggestion. I'll add it to the RFC